File: selection-sort/selectionSort.js
function findSmallest(arr) {
  smallest = arr[0];
  smallestIndex = 0;

  for(let i = 0; i < arr.length ; i++){
    if(arr[i] < smallest){
      smallest = arr[i];
      smallestIndex = i;
    }

  }
  return smallestIndex;

}

function selectionSort(arr){
  let newArr = [];
  for(let i = 0; i < arr.length; i ++) {
    let smallest = findSmallest(arr);
    let arrn = (arr.splice(smallest,1)).pop() 
    newArr.push(arrn) 
  }
  return newArr;
}

let arr = [1,5, -20 ,3,4,5,9,0, -7];
console.log(selectionSort(arr))

File: selection-sort/selection_sort.rb
def find_smallest(arr)
  smallest = arr[0]
  smallest_index = 0
 for i in (0...(arr.length)) do 
  if arr[i] < smallest
    smallest = arr[i]
    smallest_index = i
   end
 end
  return smallest_index
end

def selection_sort(arr)
  new_arr = []

  for i in (0...arr.length) do
    smallest = find_smallest(arr)
  new_arr << (arr.delete_at(smallest))
  end
return new_arr
end
arr = [3, 2, 1, 5, 7, 6, -7]

# print(selection_sort(arr))
